pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -102,8 +103,16 @@ type Results struct {
 	PodErrors     map[*v1.Pod]error
 }
 
-func (r Results) AllPodsScheduled() bool {
-	return len(r.PodErrors) == 0
+// AllNonPendingPodsScheduled returns true if all of the non-pending pods scheduled.  This is useful in consolidation as
+// we don't care if a pod was pending before consolidation and will still be pending after. It may be a pod that we
+// can't schedule at all and don't want it to block consolidation.
+func (r Results) AllNonPendingPodsScheduled() bool {
+	for p := range r.PodErrors {
+		if !pod.IsProvisionable(p) {
```

Comment:
Does this go far enough? What if a pod can't be scheduled due to an impossible affinity rule?